internal/Restore-DBFromFilteredArray.ps1

```diff
@@ -351,7 +362,7 @@ Function Restore-DBFromFilteredArray {
                     Write-Message -Level Verbose -Message "Failed, Closing Server connection"
                     $RestoreComplete = $False
                     $ExitError = $_.Exception.InnerException
-                    Write-Warning "$FunctionName - $ExitError" -WarningAction stop
+                    Stop-Function -Message  "Failed to restore db $DbName, stopping" -InnerErrorRecord $ExitError 
                     #Exit as once one restore has failed there's no point continuing
                     break
```

Comment:
Can't use it like this: Stop-Function does not accept exceptions, just error records (hence the name)
Also please avoid `break` if it isn't necessary (even though it doesn't do harm here)
Options:
```PowerShell
Stop-Function -Message  "Failed to restore db $DbName, stopping" -ErrorRecord $_
return
```
or
```PowerShell
$record = New-Object System.Management.Automation.ErrorRecord($_.Exception.InnerException, "manual", "InvalidOperation", $DbName)
Stop-Function -Message  "Failed to restore db $DbName, stopping" -ErrorRecord $record
return
```
If you must have the inner exception as the official record.